pkg/obfuscator/regex.go

```diff
@@ -0,0 +1,48 @@
+package obfuscator
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+type regexObfuscator struct {
+	ReplacementReporter
+	pattern  *regexp.Regexp
+	location string
+}
+
+func (r *regexObfuscator) FileName(s string) string {
+	if r.location == "All" || r.location == "FileName" {
+		return r.replace(s)
+	}
+	return s
+}
+
+func (r *regexObfuscator) Contents(s string) string {
+	if r.location == "All" || r.location == "FileContents" {
+		return r.replace(s)
+	}
+	return s
+}
+
+func (r *regexObfuscator) replace(input string) string {
+	output := input
+	matches := r.pattern.FindAllString(input, -1)
+	for _, m := range matches {
+		output = strings.ReplaceAll(output, m, strings.Repeat("x", len(m)))
+	}
```

Comment:
you're missing the reporting of the replacement

Reply:
Updated